docs/decisions/0039-samples-reesctructure.md

```diff
@@ -0,0 +1,652 @@
+---
+# Reestructure of How Sample Code will be Structured In the Repository
+
+status: accepted
+contact: rogerbarreto
+date: 2024-04-18
+deciders: rogerbarreto, markwallace-microsoft, sophialagerkranspandey, matthewbolanos
+consulted: dmytrostruk, sergeymenshik, westey-m, eavanvalkenburg
+informed:
+---
+
+## Context and Problem Statement
+
+- The current way the samples are structured are not very informative and not easy to be found.
+- Numbering in Kernel Syntax Examples lost its meaning.
+- Naming of the projects don't sends a clear message what they really are.
+- Folders and Solutions have `Examples` suffixes which are not necessary as everything in `samples` is already an `example`.
+
+### Current identified types of samples
+
+| Type              | Description                                                                                              |
+| ----------------- | -------------------------------------------------------------------------------------------------------- |
+| `Getting started` | A single step-by-step tutorial to get started                                                            |
+| `Concepts`        | A concept by feature specific code snippets                                                              |
+| `LearnResources`  | Code snippets that are related to online documentation sources like Microsoft Learn, DevBlogs and others |
```

Comment:
We have `LearnResources` and `Getting started` with inconsistent casing and using a space in the name, can we fix this

Reply:
Also consider `LearningResources`.  I think it's more typical.